Move AON, AONPT counter and QMSI RTC to unified interface

SavinayDharmappa opened the pull request:
This is a series of patch set which moves aon, aonpt and rtc_qmsi driver api to proposed unifying interface in PR #8340 . It also changes existing test applications to work with unified interface.

SavinayDharmappa committed:
drivers: counter: Move AON and AONPT driver api to unified interface

patch modifies AON and AONPT driver api to unified interface

Signed-off-by: Savinay Dharmappa <savinay.dharmappa@intel.com>

nordic-krch commented on drivers/counter/counter_qmsi_aonpt.c:
```diff
@@ -108,8 +109,7 @@ static u32_t aon_timer_qmsi_read(struct device *dev)
 }
 
 static int aon_timer_qmsi_set_alarm(struct device *dev,
```
legacy set_alarm function has been changed to set_wrap_alarm so this function should actually implement interface

```
typedef int (*counter_api_set_wrap_alarm)(struct device *dev,
					  u32_t ticks,
					  counter_wrap_callback_t callback,
					  void *user_data);

set_alarm should be used if device has more compare channels and alarms are set within counter period (which is set by set_wrap_alarm).
```

SavinayDharmappa replied:
ok . I change it.

nordic-krch commented on drivers/counter/counter_qmsi_aonpt.c:
```diff
@@ -108,8 +109,7 @@ static u32_t aon_timer_qmsi_read(struct device *dev)
 }
 
 static int aon_timer_qmsi_set_alarm(struct device *dev,
```
legacy set_alarm function has been changed to set_wrap_alarm so this function should actually implement interface

```
typedef int (*counter_api_set_wrap_alarm)(struct device *dev,
					  u32_t ticks,
					  counter_wrap_callback_t callback,
					  void *user_data);

set_alarm should be used if device has more compare channels and alarms are set within counter period (which is set by set_wrap_alarm).
```

SavinayDharmappa replied:
ok . I will do it.

nordic-krch commented on drivers/counter/counter_qmsi_aonpt.c:
```diff
@@ -17,9 +17,10 @@
 static void aonpt_int_callback(void *user_data);
 
-static counter_callback_t user_cb;
+static counter_alarm_callback_t user_cb;
```
it should be counter_wrap_callback_t

SavinayDharmappa replied:
ok

nashif commented:
@SavinayDharmappa can you please address the comments?

SavinayDharmappa commented:
@nordic-krch  I have addressed ur review comments.

SavinayDharmappa commented:
sanity is failing as i haven't included counter.h

nordic-krch commented on drivers/counter/counter_qmsi_aonpt.c:
```diff
@@ -108,8 +109,9 @@ static u32_t aon_timer_qmsi_read(struct device *dev)
-				    u32_t count, void *user_data)
+				    u32_t ticks,
+				    counter_wrap_callback_t callback,
+				    void *user_data)
```
note that new api does not require to set wrap only when counter is already started. So `if (QM_AONC[QM_AONC_0]->aonpt_cfg == 0)` can be removed.  Basically, it should be possible to setup the counter and then start it.

SavinayDharmappa replied:
done

nordic-krch commented:
please fix conflicts

nashif commented:
Please rebase against topic-counter branch and resubmit. Remove already merged patches.

nordic-krch commented:
please rebase on topic-counters

nashif commented:
rebased...

carlescufi commented:
@nordic-krch this is ready for re-review

SavinayDharmappa commented:
@nashif do I need to rebase it again and push it

nashif commented:
> @nashif do I need to rebase it again and push it

no

SavinayDharmappa commented on drivers/counter/counter_rtc_qmsi.c:
```diff
@@ -8,6 +8,7 @@
+#include <counter.h>
 #include <init.h>
 #include <kernel.h>
 #include <rtc.h>
```
I will remove it

SavinayDharmappa commented on drivers/counter/Kconfig:
```diff
@@ -20,6 +20,13 @@ config COUNTER_LOG_LEVEL
 	help
 	  Counter logging level.
 
+config RTC_0_NAME
```
done

SavinayDharmappa commented on samples/drivers/rtc/src/main.c:
```diff
@@ -7,35 +7,32 @@
 
 #include <device.h>
+#include <counter.h>
 #include <rtc.h>
```
will remove it

SavinayDharmappa committed:
drivers: rtc_qmsi: Move rtc driver apis to unified interface

patch modifies rtc apis to unified interface.

Signed-off-by: Savinay Dharmappa <savinay.dharmappa@intel.com>

SavinayDharmappa commented:
@nordic-krch  I am getting  "warning: ‘counter_callback_t’ is deprecated" even though i am not using counter_callback_t in driver. Can you review my patch

dbkinder reviewed (commented):
no doc changes for me to review.

SavinayDharmappa commented on soc/x86/intel_quark/quark_se/Kconfig.defconfig.series:
```diff
@@ -148,11 +150,6 @@ config WDT_QMSI
 	default y
 endif # WATCHDOG
 
-if RTC
```
will change it.

SavinayDharmappa commented on soc/x86/intel_quark/quark_d2000/Kconfig.defconfig.series:
```diff
@@ -56,16 +56,6 @@ config WDT_QMSI
 	default y
 endif # WATCHDOG
 
-if RTC
```
will change it.

SavinayDharmappa commented on tests/drivers/aon_counter/aon_api/src/test_aon_periodic_timer.c:
```diff
@@ -68,16 +68,18 @@ static int test_timer(void)
 	/* verify counter_set_alarm() */
-	if (counter_set_alarm(aon_timer, aon_timer_callback,
-			      ALARM_CNT, (void *)&dummy_data)) {
+	if (counter_set_wrap(aon_timer, ALARM_CNT,
```
will work towards it.

SavinayDharmappa committed:
soc: x86: Remove RTC config

remove using RTC config

Signed-off-by: Savinay Dharmappa <savinay.dharmappa@intel.com>

SavinayDharmappa committed:
rtc: rename CONFIG_RTC_0_NAME to DT_RTC_0_NAME.

patch renames CONFIG_RTC_0_NAME to DT_RTC_0_NAME.

Signed-off-by: Savinay Dharmappa <savinay.dharmappa@intel.com>

SavinayDharmappa committed:
tests: power: multicore: lmt: Modify app to use unified counter apis

patch modifies the app to use unified counter interface for
the test application

Signed-off-by: Savinay Dharmappa <savinay.dharmappa@intel.com>

nashif committed:
tests: remove aon_counter test

This is now covered by counter API.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>
